benchmarks/executor/test_delayed.py

```diff
@@ -80,6 +91,10 @@ def get_result_buffers(self):
             ),
         }
 
+    # Make sure we don't have default merge so that there is no default dask_merge
```

Comment:
Do we need to decouple the "can use default merge" and "can use dask_merge" cases? Can we? Do we need to document the implication that implementing a `dask_merge` means you also need to implement a normal `merge`?

Reply:
I'm working on that switching logic at the moment. Good point that one might only need a `merge()` _or_ a `merge_all()` to do the job for both cases. I'll see if I can make sth work. The point here was that for benchmarking the back-end I wanted to make sure to have a defined setup, since we'll likely implement `merge_all()` for many UDFs

Reply:
Running UDFs that only do `merge_all()` with normal executors could possibly be implemented, but that can maybe be left for later if a real need become apparent.

However, if a UDF is only being used with the DelayedJobExecutor, one doesn't have to implement `merge()`: In the default nav merge case it can use the default implementation for both variants, i.e. nothing to implement. For other cases one can only override `merge_all()`, and running with a normal executor will raise a `NotImplementedError` from the default merge.